centrallix/tests/test_argon2id_00.to

```diff
@@ -0,0 +1,12 @@
+##NAME argon2id() test
+
+query select 'case   1' = argon2id("password", "thisisasalt")
+query select 'case   2' = argon2id("password", "differentsalt")
+query select 'case   3' = argon2id("awayhaulaway", "wellhaulawayjoe")
+query select 'case   4' = argon2id("aaaaaaaaaaaaaaaa", "aaaaaaaaaaaaaaaa")
+query select 'case   5' = argon2id("P455w0rdW1th-D1ff3r3n5t!Ch@r@ct3r5??", "sodiumchloride")
+query select 'case   6' = argon2id("111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111", "111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111")
+query select 'case   7' = argon2id(":^:#@)$!#@(*&!#)$", "!@)*#&!@(*#&)!(*@&#)(!*@&#)(*")
+query select 'case   8' = argon2id("", "thissalthasanemptypassword")
+query select 'case   9' = argon2id("        ", "spacethefinalfrontier")
+query select 'case  10' = argon2id("		", "putitonmytab")
```

Comment:
Let's add some tests for the new complexity parameters, and let's add some tests for corner cases like null parameters (null/nonexistent complexity params should force a default value, but null data or salt should return null).

Also if we could include a couple of the standard tests from https://github.com/P-H-C/phc-winner-argon2/blob/master/src/test.c (see the last group of tests for argon2id) that would be great.  Note that those test calls generate multiple outputs, only one of which is the standard hexstring output.

Reply:
Ah, okay. I had simply been using `return -1` and putting up an error message when null password or null salt occurs. To clarify: you're saying you'd like me to store a null value in tree->String when that happens?

Reply:
If the data or salt is not supplied at all, that's an error condition.  If either are null, go ahead and return null, as that is typical for SQL functions.  To give a null result, just set tree->DataType to DATA_T_STRING and tree->Flags to include EXPR_F_NULL, and then return 0.

Reply:
Got it. Thanks!